src/components/ChattingRoom/ChatInput.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect, useRef, ChangeEvent, KeyboardEvent } from "react";
+import { useParams } from "react-router-dom";
+import useChatStore from "../../stores/ChatStore";
+import { newDate, getCurrentTime } from "../hooks/useTime";
+import { useCurrentUserId } from "../hooks/useUser";
+import { useMobile } from "../hooks/useMobile";
+import styled from "styled-components";
+import { Plus, Emoji, HashTag, Send } from "../../assets/icons";
+
+export function ChatInput({ onNewChat }: { onNewChat: (id: number) => void }) {
+  const { currentUserId } = useCurrentUserId();
+  const isMobile = useMobile();
+  const addChat = useChatStore((state) => state.addChat);
+  const { roomId } = useParams<{ roomId: string }>();
+
+  // useRef를 활용하여 textarea의 DOM 요소에 접근, 사용자가 입력한 텍스트 길이에 따라 textarea의 높이 조절
+  const [value, setValue] = useState<string>("");
+  const [isChanged, setIsChanged] = useState<boolean>(false);
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    const textarea = textareaRef.current;
+    if (textarea) {
+      textarea.style.height = "1.5rem";
+      textarea.style.height = `${textarea.scrollHeight}px`;
+
+      if (textarea.scrollHeight > 24) {
+        setIsChanged(true);
+      } else {
+        setIsChanged(false);
+      }
+    }
+  }, [value]);
+
+  const handleChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setValue(e.target.value);
+  };
+
+  const handleSend = () => {
+    if (!value.trim() || !currentUserId || !roomId) return; // 입력값이나 현재 유저 id나 채팅방 id가 없으면 전송 X
+
+    const newChat = {
+      id: Date.now(),
+      senderId: currentUserId,
+      text: value,
+      time: getCurrentTime(),
+    };
+
+    addChat(Number(roomId), newChat, newDate);
+    setValue("");
+
+    if (textareaRef.current) {
+      textareaRef.current.focus();
+    }
+
+    onNewChat(newChat.id);
```

Comment:
```suggestion
      textareaRef.current?.focus();
```

요렇게하면 렌더링을 줄일 수 있지 않을까 싶습니당!!